### PR TITLE
Support all image sampler kinds with brush_image implementation.

### PR DESCRIPTION
**Pull request by glennw**
As a follow up, we need to add better test coverage of these other
image types via a rawtest.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/webrender/2412)

### CODE REVIEW

**Comment by glennw** [comment]
I haven't tested this in a try build yet - will be able to kick one off in an hour or so.

r? @kvark cc @staktrace

File: webrender/res/brush_image.glsl [hunk 1]
```diff
@@ -46,7 +52,7 @@ void brush_vs(
 vec4 brush_fs() {
     vec2 uv = clamp(vUv.xy, vUvBounds.xy, vUvBounds.zw);
 
-    vec4 color = texture(sColor0, vec3(uv, vUv.z));
+    vec4 color = TEX_SAMPLE(sColor0, vec3(uv, vUv.z));
```
Comment (kvark) [inline webrender/res/brush_image.glsl]: Actually, this would be incorrect, given that we want to use the mipmaps, and this macro always picks lod of 0.0

Comment (glennw) [reply webrender/res/brush_image.glsl]: Good catch, fixed.

**Comment by kvark** [comment]
Thanks Glenn! I'd like to see a try push, unless @kats wants to get this in quickly and bear with consequences :)

**Comment by staktrace** [comment]
At the moment there's a big backlog of windows jobs in gecko CI, so I'm ok with landing this without a try push. If anything turns up I'll let you know.

**Pull request merged**